app.ts

```diff
@@ -0,0 +1,292 @@
+			console.log(error);
+		});
+}
+
+// Gets Total Of All Records
+function getRecordCount(): Promise<void> {
+	return fetch("http://localhost:2050/recordCount", {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	})
+		.then(handleErrors)
+		.then((res: Response) => res.json())
+		.then((data: number) => {
+			recordCount = data;
+		})
+		.catch((error: Error) => {
+			console.log(error);
+		});
+}
+
+//// Debounce
+function debounce(fn: any, delay: number) {
+	return function () {
+		clearTimeout(timer);
+		timer = setTimeout(() => {
+			fn();
+		}, delay);
+	};
+}
+
+//// Sizing And Resizing
+function resizing() {
+	let end = fromParameter + getNoOfRows();
+	let toParameter: number;
+	let maxRecordsID = recordCount - 1;
+
+	if (end > maxRecordsID) {
+		toParameter = maxRecordsID;
+		fromParameter = toParameter - getNoOfRows();
+	}
+	getTable();
+}
+
+//// Navigation
+// Next
+class Next {
+	nextCount: number;
+	nextButton: HTMLElement | null;
+
+	constructor() {
+		this.nextCount = 0;
+		this.nextButton = document.getElementById("next");
+
+		const nextDebounce = (fn: any, delay: number) => {
+			return () => {
+				this.nextCount++;
+				clearTimeout(timer);
+				timer = setTimeout(() => {
+					fn();
+				}, delay);
+			};
+		};
+
+		let next = () => {
+			let toParameter = getParameters(fromParameter);
+			let maxRecordsID = recordCount - 1;
+
+			if (toParameter === maxRecordsID) {
+				alert("You have reached the final page");
+			}
+
+			let nextAmount = toParameter - fromParameter + 1;
+			let nextCountAmount = nextAmount * this.nextCount;
+			fromParameter = fromParameter + nextCountAmount;
+			toParameter = fromParameter + getNoOfRows();
+
+			let end = fromParameter + getNoOfRows();
+
+			if (end > maxRecordsID) {
+				toParameter = maxRecordsID;
+				fromParameter = toParameter - getNoOfRows();
+			}
+
+			this.nextCount = 0;
+
+			getTable();
+		};
+
+		if (this.nextButton) {
+			this.nextButton.addEventListener("click", nextDebounce(next, 500));
+		}
+	}
+}
+
+// Previous
+class Prev {
+	prevCount: number;
+	prevButton: HTMLElement | null;
+
+	constructor() {
+		this.prevCount = 0;
+		this.prevButton = document.getElementById("prev");
+
+		const prevDebounce = (fn: any, delay: number) => {
+			return () => {
+				this.prevCount++;
+				clearTimeout(timer);
+				timer = setTimeout(() => {
+					fn();
+				}, delay);
+			};
+		};
+
+		let prev = () => {
+			let toParameter = getParameters(fromParameter);
+
+			if (fromParameter === 0) {
+				alert("You Are On The First Page");
+			} else {
+				let prevAmount = toParameter - fromParameter + 1;
+				let prevCountAmount = prevAmount * this.prevCount;
+
+				let intOne = fromParameter - prevCountAmount;
+
+				if (intOne < 0) {
+					fromParameter = 0;
+				} else {
+					fromParameter = intOne;
+				}
+
+				this.prevCount = 0;
+
+				getTable();
+			}
+		};
+
+		if (this.prevButton) {
+			this.prevButton.addEventListener("click", prevDebounce(prev, 500));
```

Comment:
Same comment about the debounce that I made on the `Next` class, should help you get rid of the `prevCount`.